tests/reference/asr-test_assign9-51278b8.stderr

```diff
@@ -2,4 +2,4 @@ semantic error: Assignment to an input function parameter `x` is not allowed
  --> tests/errors/test_assign9.py:2:5
   |
 2 |     x = 2
-  |     ^ Use InOut[x] to allow assignment
+  |     ^ Assignment not supported for `x`
```

Comment:
For arrays, structs, etc the most common reason to get this error is when trying to modify the value in the caller. So the best suggestion is to use InOut (or Out).

For integers, people know that you cannot modify the variable in the caller that way, so I don't think that's the most common reason to get this error message.

I think there are two possibilities why somebody would assign to `x`:

* Trying to use the variable `x` locally in some algorithm and simply modifying it in the process, knowing that it doesn't change in the caller;
* Not needing the argument anymore and wanting to use a new local variable that happens to have the same name
* By a mistake wanting to use a new local variable that happens to shadow the argument, causing a bug

```suggestion
  |     ^ create a new local variable with a different name
```